src/main/groovy/life/qbic/datamodel/dtos/business/services/Product.groovy

```diff
@@ -54,15 +78,54 @@ abstract class Product {
    * @param unitPrice The price in € per unit
    * @param unit The product unit
    * @param productId The Id of the product
+   *
+   * @deprecated 2.11.0
    */
+  @Deprecated
   Product(String name, String description, double unitPrice, ProductUnit unit, ProductId productId) {
     this.productName = Objects.requireNonNull(name, "Name must not be null")
     this.description = Objects.requireNonNull(description, "Description must not be null")
-    this.unitPrice = Objects.requireNonNull(unitPrice, "Unit price must not be null")
+    this.unitPrice = Objects.requireNonNull(unitPrice, "Internal unit price must not be null")
     this.unit = Objects.requireNonNull(unit, "Unit must not be null")
     this.productId = Objects.requireNonNull(productId, "ProductId must not be null")
     //currency is on default in euro
     this.currency = Currency.getInstance(Locale.GERMANY)
+    this.externalUnitPrice = 0.00
+    this.internalUnitPrice = 0.00
```

Comment:
`unitPrice` == `internalUnitPrice` ? Here code and JavaDoc do not match.

Reply:
in case the old constructor is used, these properties have to be 0